source/vibe/data/bson.d

```diff
@@ -1428,7 +1428,7 @@ struct BsonSerializer {
 			m_type = tp;
 		}
 
-		if (m_entryName) {
+		if (m_entryName != "") {
```

Comment:
Still not perfect, but this should better be `!is null`, because dictionary keys can in theory be `""`, too.